nixos/modules/config/users-groups.nix

```diff
@@ -693,7 +693,20 @@ in {
               users.groups.${user.name} = {};
             '';
           }
-        ]
+        ] ++ (map (shell: {
+            assertion = (user.shell == pkgs.${shell}) -> (config.programs.${shell}.enable == true);
+            message = ''
+              users.users.${user.name}.shell is set to ${shell}, but
+              programs.${shell}.enable is not true. This will cause the ${shell}
```

Comment:
Maybe include "in your nixos configuration" here to clarify that it's not enough to enable your shell from home manager? (Although I'm curious if this assertion is fixing real world problems. Was a lot of users reporting problems with having shell set but it not being enabled?

Reply:
The original reporting user was myself, but I broke fish :) it was pretty annoying as the login shell lost all the $PATH.